packages/core/src/transport-http/transport-http.ts

```diff
@@ -479,6 +593,65 @@ export class StreamableHttpTransport {
     }
   }
 
+  private async handleBatchRequest(
+    batch: unknown[],
+    sessionId: string | null,
+    options?: { authInfo?: AuthInfo },
+  ): Promise<Response> {
+    const responses: JsonRpcRes[] = [];
+
+    for (const message of batch) {
+      if (!isJsonRpcRequest(message) && !isJsonRpcNotification(message)) {
+        // Invalid message in batch
+        responses.push(
+          createJsonRpcError(
+            null,
+            new RpcError(
+              JSON_RPC_ERROR_CODES.INVALID_REQUEST,
+              "Invalid JSON-RPC 2.0 message in batch",
+            ).toJson(),
+          ),
+        );
+        continue;
+      }
+
+      try {
+        const response = await this.server?._dispatch(message, {
+          sessionId: sessionId || undefined,
+          sessionProtocolVersion:
+            await this.getSessionProtocolVersion(sessionId),
```

Comment:
nit: i would put this `await` at the top, otherwise you're hitting the session store for each message in the batch